src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,132 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.exception.LoadingError;
+import duke.exception.SavingError;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Class that handles laoding and saving tasks to a file.
+ */
+public class Storage {
+
+    private final String filepath;
+
+    /**
+     * Storage constructor.
+     *
+     * @param filepath The filepath of the task list from the project root
+     */
+    public Storage(String filepath) {
+        this.filepath = filepath;
+    }
+
+    /**
+     * Returns an ArrayList<Task> of loaded Tasks from the file.
+     *
+     * @return ArrayList<Task> of tasks
+     * @throws DukeException that indicates what the issue was in opening/parsing the file
+     */
+    public ArrayList<Task> load() throws DukeException {
+        String home = System.getProperty("user.dir");
+        Path path = Paths.get(home, this.filepath);
+        boolean directoryExists = Files.exists(path);
+
+        if (directoryExists) {
+            try {
+                List<String> lines = Files.readAllLines(path);
+                ArrayList<Task> tasks = new ArrayList<>();
+                for (String x: lines) {
+                    String[] data = x.split(Pattern.quote(" | "));
+                    Task t;
+                    if (data.length < 2) {
+                        continue;
+                    }
+                    switch (data[1]) {
+                    case "T":
+                        t = new ToDo(data[3]);
+                        break;
+                    case "E":
+                        LocalDate date = LocalDate.parse(data[4]);
+                        if (data.length == 5) {
+                            t = new Event(data[3], date);
+                        } else {
+                            String hour = data[5].replace(":", "").substring(0, 2);
+                            String minute = data[5].replace(":", "").substring(2, 4);
+                            LocalTime time = LocalTime.of(Integer.parseInt(hour), Integer.parseInt(minute));
+                            t = new Event(data[3], date, time);
+                        }
+                        break;
+                    case "D":
+                        LocalDate date2 = LocalDate.parse(data[4]);
```

Comment:
Maybe can name as deadlineDate and eventDate for easy reference, as date and date2 can be confusing?